File: src/pack/Scenerio14.scala
package pack
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.expressions._
object Scenerio14 {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("Scenerio14")
    val sc = new SparkContext(conf)
    sc.setLogLevel("ERROR")
    val spark = SparkSession.builder().getOrCreate()
    import spark.implicits._

    val df = Seq((203040, "rajesh", 10, 20, 30, 40, 50)).toDF("rollno", "name", "telugu", "english", "maths", "science", "social")
    df.show()

    //Through SQL
    df.createOrReplaceTempView("marks")
    spark.sql("select *, (telugu+english+maths+science+social) as total from marks").show()

    //Through DSL
    val finaldf = df.withColumn("total", expr("telugu+english+maths+science+social")).show()
  }
}

File: Scenerio14.py
from pyspark import SparkConf, SparkContext
from pyspark.sql import SparkSession
from pyspark.sql import *
from pyspark.sql.types import *
from pyspark.sql.functions import *
from pyspark.sql.window import *

conf = SparkConf().setMaster("local[*]").setAppName("Scenerio14")
sc = SparkContext(conf=conf)
sc.setLogLevel("ERROR")
spark = SparkSession.builder.getOrCreate()

data = [
    (203040, "rajesh", 10, 20, 30, 40, 50)
]

df = spark.createDataFrame(data, ["rollno", "name", "telugu", "english", "maths", "science", "social"])
df.show()

# Through SQL
df.createOrReplaceTempView("marks")
spark.sql("select *, (telugu+english+maths+science+social) as total from marks").show()

# Through DSL
finaldf = df.withColumn("total", expr("telugu+english+maths+science+social")).show()
